### PR TITLE
Add displayProductPageDrawer hook

### PR DESCRIPTION
**Pull request by antoin-m**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.1.x
| Description?  | Add displayProductPageDrawer hook
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |

The module has to return something like 

```php
    public function hookDisplayProductPageDrawer($params)
    {
        return array(
            new \PrestaShop\PrestaShop\Core\Product\ProductAdminDrawer(array(
            'title' => 'stats',
            'link' => 'http://localhost/index.html',
            'id' => 'ps_banner',
            'icon' => 'label',
       )));
    }
```
or
```php
    public function hookDisplayProductPageDrawer($params)
    {
        return array(
            // The value can be send via an array for the constructor
            new \PrestaShop\PrestaShop\Core\Product\ProductAdminDrawer(array(
            'title' => 'Stats',
            'link' => 'http://localhost/index.html',
            'id' => 'ps_banner',
            'icon' => 'help',
        )), // Or with setters for each properties
            (new \PrestaShop\PrestaShop\Core\Product\ProductAdminDrawer())
            ->setTitle('Plan')
            ->setLink('http://localhost/index.html')
        );
    }
```
Result:
capture du 2017-03-13 17-25-40

### CODE REVIEW

File: install-dev/upgrade/sql/1.7.1.1.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
+  (NULL, 'displayProductPageDrawer', 'Product Page Drawer', 'This hook displays content in the right sidebar of the product page', '1');
```
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.1.1.sql]: If it's for 1.7.1.0, it's not good

**Review by aleeks — CHANGES_REQUESTED** [review]
See comment

File: install-dev/upgrade/sql/1.7.1.1.sql (hunk 1 above)
Comment (antoin-m) [inline install-dev/upgrade/sql/1.7.1.1.sql]: Didn't we say 1.7.1.1 ? @vincentbz

Comment (aleeks) [inline install-dev/upgrade/sql/1.7.1.1.sql]: Don't know :'(

Comment (vincentbz) [inline install-dev/upgrade/sql/1.7.1.1.sql]: Na, we said 1.7.1.0 as addons needs it :p

Comment (aleeks) [inline install-dev/upgrade/sql/1.7.1.1.sql]: So 1.7.1.0 baby !

**Review by aleeks — CHANGES_REQUESTED** [review]
According with @vincentbz

**Comment by Quetzacoalt91** [comment]
Hoy,

I see we expect a specific structure here.
If you want to be sure a module developer will follow it, it would be better to create a class in which you write all the properties available, and then you use `PrestaShopBundle\Service\Hook\HookFinder` to get an array of them.

Why would you do that ? It would prevent you to check again if the properties exists, and the module id will be auto-populated when you get the presented data (https://github.com/PrestaShop/PrestaShop/blob/d9c37b1bfff26e43ea9df3728fdeca4041c47c84/src/PrestaShopBundle/Service/Hook/HookFinder.php#L112). :)

**Review by Quetzacoalt91 — DISMISSED** [review]
Hoy,

I see we expect a specific structure here.
If you want to be sure a module developer will follow it, it would be better to create a class in which you write all the properties available, and then you use `PrestaShopBundle\Service\Hook\HookFinder` to get an array of them.

Why would you do that ? It would prevent you to check again if the properties exists, and the module id will be auto-populated when you get the presented data (https://github.com/PrestaShop/PrestaShop/blob/d9c37b1bfff26e43ea9df3728fdeca4041c47c84/src/PrestaShopBundle/Service/Hook/HookFinder.php#L112). :)

**Comment by aleeks** [comment]
I think, we have to do another PR, @antoin-m is so busy at this time :cry: 
I'm closing this one... :(

**Comment by antoin-m** [comment]
Hahaha sorry 😥

**Comment by Quetzacoalt91** [comment]
@aleeks If we close this PR, we will forget the feature we want to add to the software.

Of course @antoin-m may be too busy to update the pull-request, but we can do it ... if the feature worth it. :)

**Comment by kpodemski** [comment]
It's worth it

**Review by aleeks — DISMISSED** [review]

**Comment by Quetzacoalt91** [comment]
The pull-request has been updated to use the class `HookFinder`.

**Comment by vincentbz** [comment]
Thanks @Quetzacoalt91 !
I test it and then we can merge it ?

**Comment by Quetzacoalt91** [comment]
Yep we can if a code review pass

**Comment by maximebiloe** [comment]
Everything looks fine to me.
Thanks guys!

**Comment by antoin-m** [comment]
image

**Comment by xBorderie** [comment]
@antoin-m Tut tut tut, don't forget the last one: https://github.com/PrestaShop/PrestaShop/pulls/antoin-m :)

**Comment by antoin-m** [comment]
image

The ball is in your court 🥇

**Pull request merged**